dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,596 @@
+ * <ul>
+ * <li>A new column called {@code scheme_id} will be added to the {@code workflow_action} table.
+ * This allows actions to be assigned to Workflow Schemes and not to steps directly.</li>
+ * <li>The new {@code workflow_action_step} table allows an action to be associated to one or more
+ * steps (an N-N relationship).</li>
+ * <li>The index created for the {@code workflow_scheme_x_structure} column will be updated to NOT
+ * be unique.</li>
+ * </ul>
+ *
+ * @author Jose Castro
+ * @version 4.3.0
+ * @since Nov 1st, 2017
+ */
+public class Task04305UpdateWorkflowActionTable implements StartupTask {
+
+    private static final String MYSQL_FIND_INTERMEDIATE_TABLE    = "SELECT * from workflow_action_step";
+    private static final String POSTGRES_FIND_INTERMEDIATE_TABLE = "SELECT * from workflow_action_step";
+    private static final String MSSQL_FIND_INTERMEDIATE_TABLE    = "SELECT * from workflow_action_step";
+    private static final String ORACLE_FIND_INTERMEDIATE_TABLE   = "SELECT * from workflow_action_step";
+
+    private static final String MYSQL_CREATE_INTERMEDIATE_TABLE = "CREATE TABLE workflow_action_step (action_id VARCHAR(36) NOT NULL, step_id VARCHAR(36) NOT NULL, action_order INT default 0)";
+    private static final String POSTGRES_CREATE_INTERMEDIATE_TABLE = MYSQL_CREATE_INTERMEDIATE_TABLE;
+    private static final String MSSQL_CREATE_INTERMEDIATE_TABLE = "CREATE TABLE workflow_action_step ( action_id NVARCHAR(36) NOT NULL, step_id NVARCHAR(36) NOT NULL action_order INT default 0, CONSTRAINT pk_workflow_action_step PRIMARY KEY NONCLUSTERED (action_id, step_id) )";
+    private static final String ORACLE_CREATE_INTERMEDIATE_TABLE = "CREATE TABLE workflow_action_step ( action_id VARCHAR(36) NOT NULL, step_id VARCHAR(36) NOT NULL, action_order number(10,0) default 0, CONSTRAINT pk_workflow_action_step PRIMARY KEY (action_id, step_id) )";
+
+    private static final String MYSQL_CREATE_INTERMEDIATE_TABLE_PK = "ALTER TABLE workflow_action_step ADD CONSTRAINT pk_workflow_action_step PRIMARY KEY (action_id, step_id)";
+    private static final String POSTGRES_CREATE_INTERMEDIATE_TABLE_PK = MYSQL_CREATE_INTERMEDIATE_TABLE_PK;
+
+    private static final String MYSQL_FIND_REQUIRES_CHECKOUT_OPTION_COLUMN     = "SELECT requires_checkout_option FROM workflow_action";
+    private static final String POSTGRES_FIND_REQUIRES_CHECKOUT_OPTION_COLUMN  = "SELECT requires_checkout_option FROM workflow_action";
+    private static final String MSSQL_FIND_REQUIRES_CHECKOUT_OPTION_COLUMN     = "SELECT requires_checkout_option FROM workflow_action";
+    private static final String ORACLE_FIND_REQUIRES_CHECKOUT_OPTION_COLUMN    = "SELECT requires_checkout_option FROM workflow_action";
+
+    private static final String MYSQL_FIND_SCHEME_ID_COLUMN    = "SELECT scheme_id FROM workflow_action";
+    private static final String POSTGRES_FIND_SCHEME_ID_COLUMN = "SELECT scheme_id FROM workflow_action";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'MYSQL_FIND_SCHEME_ID_COLUMN' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)